azure-sql/database/performance-guidance.md

```diff
@@ -261,9 +261,14 @@ Connection pooling can improve throughput, reduce latency, and enhance the overa
    - [JDBC connection pooling](/sql/connect/jdbc/using-connection-pooling?view=azuresqldb-current&preserve-view=true)
    - [PHP connection pooling](/sql/connect/php/connection-pooling-microsoft-drivers-for-php-for-sql-server?view=azuresqldb-current&preserve-view=true)
 
-- Cloud applications should implement [retry logic](develop-overview.md#resiliency) to handle transient connectivity failures gracefully. Learn more about how to design [retry logic for transient errors](troubleshoot-common-connectivity-issues.md#retry-logic-for-transient-errors). 
+- Cloud applications should implement [retry logic](develop-overview.md#resiliency) to handle transient connectivity failures gracefully. Learn more about how to design [retry logic for transient errors](troubleshoot-common-connectivity-issues.md#retry-logic-for-transient-errors).
+
+- Token-based authentication mechanisms, such as Microsoft Entra ID authentication, can disrupt connection pooling when tokens expire, invalidating pooled connections and preventing their reuse. To address this:
+  - **Implement proactive token renewal:** [Refresh Microsoft Entra ID tokens](/entra/identity-platform/refresh-tokens) to renew access tokens *before* expiration. This minimizes interruptions to connection reuse caused by invalid tokens.
+  - **Adjust token lifetimes:** [Configure token expiration policies in Microsoft Entra ID](/entra/identity/conditional-access/howto-conditional-access-session-lifetime) to accommodate the expected duration of connections in your application. 
+
+- [Monitor Azure SQL Database](monitoring-sql-database-azure-monitor.md) connection performance and resource usage to identify bottlenecks, such as excessive idle connections, insufficient pool limits, or authentication issues caused by expired tokens, and adjust configurations accordingly. Use [Microsoft Entra ID logs](/entra/identity/monitoring-health/concept-sign-ins) or [Azure Monitor](monitoring-metrics-alerts.md) to track token expiration errors and ensure token lifetimes are appropriately configured. Consider using [database watcher](../database-watcher-overview.md) or Azure Monitor for overall connection performance monitoring.
```

Comment:
Azure Monitor metrics don't track token expiration errors, at least not on the database side. Not sure if something exists on the client side. Database watcher doesn't either because this is not exposed in any DMV in the database.

Reply:
@dimitri-furman thanks for feedback. Modified it accordingly.